resource/errors.go

```diff
@@ -84,15 +84,18 @@ func NewBuildTimeoutError(name Name) error {
 
 // DependencyNotFoundError is used when a resource is not found in a dependencies.
 func DependencyNotFoundError(name Name) error {
-	return errors.Errorf("%q missing from dependencies", name)
+	// This error represents a logical configuration error. No need to include a stack trace.
+	return fmt.Errorf("Resource missing from dependencies. Resource: %v", name)
 }
 
 // DependencyTypeError is used when a resource doesn't implement the expected interface.
 func DependencyTypeError[T Resource](name Name, actual interface{}) error {
+	// This error represents a coding error. Include a stack trace for diagnostics.
```

Comment:
I think this and the following error are the reason someone wanted stack traces. Only a few locations are using these calls, but I can see the stack trace being useful as this largely represented a coding problem.

Reply:
Yeah I buy that this and the following error are issues with model implementations and not an error in the JSON config.